Makefile

```diff
@@ -158,7 +158,10 @@ run-android: $(ATESTS)
 
 run-test-jnimarshal: bin/Test$(CONFIGURATION)/Java.Interop.Export-Tests.dll bin/Test$(CONFIGURATION)/$(JAVA_INTEROP_LIB)
 	MONO_TRACE_LISTENER=Console.Out \
-	$(RUNTIME) bin/$(CONFIGURATION)/jnimarshalmethod-gen.exe "$<"
+	$(RUNTIME) bin/$(CONFIGURATION)/jnimarshalmethod-gen.exe -v -L $(JI_MONO_PATH)mono/4.5 -L $(JI_MONO_PATH)mono/4.5/Facades "$<"
+	$(RM) "$<" "$(basename $<)-JniMarshalMethods.dll"
+	mv "$(basename $<)-new.dll" "$<"
```

Comment:
It looks like the semantics of `jnimarshalmethod-gen.exe` is that it now produces two assemblies:

* `$(basename $<)-JniMarshalMethods.dll`, and
* `$(basename $<)-new.dll`

The first seems useful as a debugging aid, though perhaps should only be preserved if `jnimarshalmethod-gen.exe -keeptemps` is used.

The second is...weird. I'd expect `jnimarshalmethod-gen.exe` to *update-and-replace* the specified file, not spit out some magical `-new.dll` assembly.

This doesn't need to be fixed in this PR, but it should be fixed.

Reply:
That would need to change the way we use the reflection as the loaded assembly locks the file, so it cannot be rewritten or deleted.

Possible solution might be to load the assembly in a separate app domain and unload the app domain once the reflection work is done. That way we can save the assembly to the originating one later and also delete the intermediate *-JniMarshalMethods assembly.